lib/dispatcher/client-h1.js

```diff
@@ -201,8 +201,6 @@ class Parser {
      * @param {*} llhttp
      */
   constructor (client, socket, { exports }) {
-    assert(Number.isFinite(client[kMaxHeadersSize]) && client[kMaxHeadersSize] > 0)
```

Comment:
I'd suggest to keep this one because of my previous point about the defaults, the rest LGTM

Reply:
we dont need to have this check because we set it already properly on client.